## Introduction

The internship at ICIPE has been an invaluable learning experience so far. These are the [milestones](Roadmap.md) I had set at the begining of the internship. You can find my progress [here](https://github.com/Asatsa/internship_repo1/issues/2)


### Introduction to bioinformatics
This covered an array of important parts in bioinformatics. This includes principal rules or practices which are essential for good research practice in bioinformatics.
#### Data Management
Data management is an important part of bioinformatics to enable the principle of open science where which encourages FAIR bioinformatics. i.e Findable, Accessible, Interoperable and Reusable. Access to data enables collaborations and reproducibility
##### Databases
I was able to learn about biological databases for example NCBI and EBI , constituting of primary,secondary and tertiary categories. I also understood their importance in ensuring access to data.

#### Version control
What version of the program and data is used to achieve a particular result. This is especially paramount to enable reproducibiliy. In particular we discussed and made use of git:

##### Git and github

Git for version control has been helpful in collaborations and just keeping track of work. I have learnt about creating and sharing projects. The practical part of using git i.e git pull and push is still lacking. I have found it difficult to resolve conflict from the terminal. This has led to a tendency of working from git hub like I am doing now ðŸ˜…. This is to say that my comfort at working with git is still lacking. I have however found git cheat sheets to very useful, here is an [example](https://www.freecodecamp.org/news/git-cheat-sheet/)

#### Reproducibility
This was one of the most important lessons, here I learnt about documentation in detail the use of notebooks which is important for enabling reproducibility and sharing. I was also introduced to workflow languages i.e snakemake and nextflow.
I was also introduced to The Galaxy project:
###### Galaxy workshop
I had an opportunity to attend a galaxy training workshop, GTN Smorgasbod organised by the Galaxy training Network. Here,I got an indepth introduction into galaxy with key highlights being genome annotation and RNAseq analysis. This experience was an eye opener as it introduced me to cloud computing which is where analysis involving big data is done efficiently. To find out more about my experience with galaxy please click [here](Galaxy report.md)


### Programming languages
### Training in Bash

This was the first programming language I learnt at ICIPE. I have learnt basic commands, navigation(directories), text editors, scripts looping e.t.c. This is are the training materials that I used with the help of an instructor[]. Furthermore I was engaged in a test to enhance my bash understanding, you can find the quiz in this [repo](https://github.com/Asatsa/bash-quiz). This is my primary programming language of performing assigned tasks.


### Training in Python
To learn basic knowledge in python, we were taken through a workshop [Introduction to pytyhon for bioinformatics](https://github.com/kipkurui/Python4Bioinformatics2019) in a formal class set up. I learnt about;    Basic data types and operations (numbers, strings); String manipulation; Data structures: Lists and Tuples and dictionarie ; control statements: if, for, while, try statements; functions ; scripting ; data analysis and plotting with Pandas and reproducible Bioinformatics Research . This was followed by individual follow up where I completed a python [carpenty](https://swcarpentry.github.io/python-novice-inflammation/). I have and I am still working on python for bioinformatics, which employs Biopthon and a set of other libraries which are for analysising biological data. [Workshop](https://github.com/peterjc/biopython_workshop). Here I am working through; reading sequence files;writing sequence files; working with sequence features and reading and writing alignment files.

### 16S- Accredetation
This was a node accredetation exercise where ICIPE MBBU was trying to get certified to perform 16s analysis this is by demonstrating our ability to analyse and interpret the data, test our analysis platform performance, and write a comprehensive report.some of the responsibilities included, developing and testing various pipelines, converting the pipelines to a workflow language (Nextflow) and report writing. 

I worked on developing a new pipeline, DADA2, with collegues @Simeone hebrew and @Kauthar Omar. This experience was eye opening. I was able to develop skills in R programming and more importantly, problem solving and/or debugging. You can find my scripts and the reports [here](https://github.com/mbbu/16S_Accreditation)

#### HPC computing
This is was introduced in part with the accredetation because the data was to be analysed on icipe hpc. This included basic navigation between the nodes e.t.c.

### Challenges
1. Git on the commandline
2. workflow languages

### Conclusion
My experience so far at ICIPE has been a learning marathon which I appreciate. I am looking forward to :
1. A mini-project to solidify these programming skills and also teach me how to make proper biological inferences from analyis of bioilogical data
2. Getting atleast conversant with nextflow
3. Go back to  git on the command line

